### PR TITLE
ci: add codespell for spell check

### PR DESCRIPTION
**Pull request by Madhu-1**
adding a GitHub action to check the spelling mistakes in the code.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
This will be a good addition to our CI

File: internal/rbd/clone.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func (rv *rbdVolume) checkCloneImage(ctx context.Context, parentVol *rbdVolume)
 		}
 		return true, nil
 	}
-	// as the temp clone doesnot exists,check snapshot exists on parent volume
+	// as the temp clone doesn't exists,check snapshot exists on parent volume
```
Comment (nixpanic) [inline internal/rbd/clone.go:113]: `does not exist` (not exists)

File: internal/rbd/controllerserver.go [hunk 2]
```diff
@@ -545,7 +545,7 @@ func checkContentSource(ctx context.Context, req *csi.CreateVolumeRequest, cr *u
 			if !errors.Is(err, ErrSnapNotFound) {
 				return nil, nil, status.Error(codes.Internal, err.Error())
 			}
-			return nil, nil, status.Errorf(codes.NotFound, "%s snapshot doesnot exists", snapshotID)
+			return nil, nil, status.Errorf(codes.NotFound, "%s snapshot does not exists", snapshotID)
```
Comment (nixpanic) [inline internal/rbd/controllerserver.go:548]: `does not exist` - again `exists` with an `s`

File: .github/workflows/codespell.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+# Referance https://github.com/codespell-project/actions-codespell
```
Comment (nixpanic) [inline .github/workflows/codespell.yaml:2]: you might want to add a little description of the test too?

File: .github/workflows/codespell.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# yamllint disable-line rule:truthy
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+    branches:
+      - "master"
```
Comment (nixpanic) [inline .github/workflows/codespell.yaml:13]: Decide if you need `"master"` with, or without quotes, and use the same everywhere

**Comment by nixpanic** [comment]
Moving simple checks from Travis CI to GitHub CI might be good. Note that **ci/centos/containerized-tests** could get extended as well.

File: Makefile [hunk 5]
```diff
@@ -150,7 +150,7 @@ e2e.test: check-env
 
 #
 # e2e testing by compiling e2e.test in case it does not exist and running the
-# executable. The e2e.test executable is not checked as a dependency in the
+# executable. The e2e.test executable is not checked as a dependencies in the
```
Comment (nixpanic) [inline Makefile:153]: `dependency` should be single here

File: Makefile [hunk 6]
```diff
@@ -191,7 +191,7 @@ else
 endif
 
 ifeq ($(USE_PULLED_IMAGE),no)
-# create a (cached) container image with dependencied for testing cephcsi
+# create a (cached) container image with dependency for testing cephcsi
```
[the remark this answers is not in the document]
Comment (nixpanic) [reply Makefile:194]: `dependencies` was corrected on line 182, the same should be done here

**Review by nixpanic — CHANGES_REQUESTED** [review]
still not all issues addressed, but almost good to go :+1:

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
Actually, lets wait with `[ready-to-merge]` until #1644 has passed CI (finally) and is merged

**Comment by Madhu-1** [comment]
Manually merging as GitHub workflow is added and Mergify cannot merge if GitHub flows are added/modified

**Pull request merged**